### PR TITLE
`flatten-array` Add additional test cases

### PR DESCRIPTION
**Pull request by kwchang0831**
2nd Proposal of #1951 

Add additional test cases to test if solution can correctly filter out null values from input list without filter out other non-null values.

### CODE REVIEW

**Review by BethanyG — APPROVED** [review]

**Comment by BethanyG** [comment]
@siebenschlaefer 

> (Disclaimer: I'm not a native English speaker.)

_Which makes you much better at catching typos!_

File: exercises/flatten-array/canonical-data.json [hunk 1]
```diff
@@ -168,6 +168,38 @@
...
+          3
+        ]
+      },
+      "expected": [
+        3
+      ]
+    },
+    {
+      "uuid": "382c5242-587e-4577-b8ce-a5fb51e385a1",
+      "description": "consecutibe null values in the middle of list are omitted from the final result",
```
Comment (BethanyG) [inline exercises/flatten-array/canonical-data.json:188]:
```suggestion
      "description": "consecutive null values in the middle of the list are omitted from the final result",
```

**Review by BethanyG — CHANGES_REQUESTED** [review]
Put in the suggestions for typo correction caught by @siebenschlaefer.

**Comment by kwchang0831** [comment]
Really appreciate all your help. I have updated all the typo and suggested changes.

**Review by BethanyG — APPROVED** [review]
Nice work!  🌟

**Comment by BethanyG** [comment]
Paging @exercism/reviewers  for additional reviews/approval.  😄

**Review by junedev — APPROVED** [review]

**Review by neenjaw — APPROVED** [review]

File: exercises/flatten-array/canonical-data.json [hunk 2]
```diff
@@ -168,6 +168,38 @@
...
+          3
+        ]
+      },
+      "expected": [
+        3
+      ]
+    },
+    {
+      "uuid": "382c5242-587e-4577-b8ce-a5fb51e385a1",
+      "description": "consecutive null values at the middle of the list are omitted from the final result",
```
Comment (petertseng) [inline exercises/flatten-array/canonical-data.json:188]:
```suggestion
      "description": "consecutive null values in the middle of the list are omitted from the final result",
```

while I think "in the middle of" seems more natural than "at the middle of", since English isn't my first language, it's very possible I am wrong.

**Review by petertseng — APPROVED** [review]

File: exercises/flatten-array/canonical-data.json (hunk 2 above)
Comment (BethanyG) [reply exercises/flatten-array/canonical-data.json:188]: I speak English as a first language, and I think either works.  But looking at it again, I agree that `in` is a bit more natural.  😄

Comment (kwchang0831) [reply exercises/flatten-array/canonical-data.json]: There is another one, if we wanted to use `in`.

Comment (kwchang0831) [reply exercises/flatten-array/canonical-data.json]: Hmm... I am a bit confused. 
(I am not native English speaker neither.)

So, the final decision would be
> in the middle of the list 
and 
> at the front of the list
?

I will make sure I avoid these grammar mistakes in the future. Thanks :D
